modules/underdogmediaBidAdapter.js

```diff
@@ -1,112 +1,116 @@
+
+  buildRequests: function (validBidRequests) {
+    spec.bidParams = validBidRequests;
     var sizes = [];
     var siteId = 0;
 
-    bidParams.bids.forEach(bidParam => {
+    spec.bidParams.forEach(bidParam => {
       sizes = utils.flatten(sizes, utils.parseSizesInput(bidParam.sizes));
       siteId = bidParam.params.siteId;
     });
-    adloader.loadScript(getJsStaticUrl + 'sid=' + siteId + ';sizes=' + sizes.join(','), null, false);
-  }
 
-  function _callback(response) {
-    var mids = response.mids;
-    bidParams.bids.forEach(bidParam => {
-      var filled = false;
-      mids.forEach(mid => {
+    return {
+      method: 'GET',
+      url: `${window.location.protocol}//udmserve.net/udm/img.fetch`,
+      data: `tid=1;dt=10;sid=${siteId};sizes=${sizes.join(',')}`,
+    };
+  },
+
+  interpretResponse: function (serverResponse, bidRequest) {
+    const bidResponses = [];
+    spec.bidParams.forEach(bidParam => {
+      serverResponse.body.mids.forEach(mid => {
         if (mid.useCount > 0) {
           return;
         }
+
         if (!mid.useCount) {
           mid.useCount = 0;
         }
+
         var size_not_found = true;
         utils.parseSizesInput(bidParam.sizes).forEach(size => {
           if (size === mid.width + 'x' + mid.height) {
             size_not_found = false;
           }
         });
+
         if (size_not_found) {
           return;
         }
 
-        var bid = bidfactory.createBid(1, bidParam);
-        bid.bidderCode = bidParam.bidder;
-        bid.width = mid.width;
-        bid.height = mid.height;
+        const bidResponse = {
+          requestId: bidParam.bidId,
+          bidderCode: spec.code,
+          cpm: parseFloat(mid.cpm),
+          width: mid.width,
+          height: mid.height,
+          ad: mid.ad_code_html,
+          creativeId: mid.mid,
+          currency: 'USD',
+          netRevenue: false,
+          ttl: config.getConfig('_bidderTimeout'),
+        };
 
-        bid.cpm = parseFloat(mid.cpm);
-        if (bid.cpm <= 0) {
+        if (bidResponse.cpm <= 0) {
           return;
         }
-        mid.useCount++;
-        bid.ad = mid.ad_code_html;
-        bid.ad = _makeNotification(bid, mid, bidParam) + bid.ad;
-        if (!(bid.ad || bid.adUrl)) {
+        if (bidResponse.ad.length <= 0) {
           return;
         }
-        bidmanager.addBidResponse(bidParam.placementCode, bid);
-        filled = true;
+
+        mid.useCount++;
+
+        bidResponse.ad += spec.makeNotification(bidResponse, mid, bidParam);
+
+        bidResponses.push(bidResponse);
       });
-      if (!filled) {
-        var nobid = bidfactory.createBid(2, bidParam);
-        nobid.bidderCode = bidParam.bidder;
-        bidmanager.addBidResponse(bidParam.placementCode, nobid);
-      }
     });
-  }
 
-  $$PREBID_GLOBAL$$.handleUnderdogMediaCB = _callback;
+    return bidResponses;
+  },
 
-  function _makeNotification(bid, mid, bidParam) {
+  makeNotification: function (bid, mid, bidParam) {
```

Comment:
`makeNotification` and `getUrlVars` function can remain private in adapter. No need to added them to `spec`